Makefile

```
@@ -77,50 +60,86 @@ CXXFLAGS =
 
 include $(LLBUILD)/Makefile.include
 
-ifdef prefix
-CONFIGURE_ARGS = --prefix=$(prefix)
-else
-CONFIGURE_ARGS = 
-endif
-
-ifeq ($(MACH),x86_64)
-all: multilib
-else
-all: regular
-endif
+# Brief build system description
+#
+# To build we use autotools but without automake
+# this roughly maps onto for example GDB or Emacs.
+# Thous the Makefile loosely follow GNU standards.
+#
+# The build follows the usual
+# ./configure
+# make
+# make install schema
+#
+# In practice that means the build system is separated into
+# three stages:
+# - bootstrap
+# - setup
+# - build
+
+# Build system files
+
+# bootstrap files generated through autoreconf.
+# Clean them resets the whole generated of the source-tree
+BOOTSTRAP_FILES = \
+	$(SRCDIR)/config.guess \
+	$(SRCDIR)/config.sub \
+	$(SRCDIR)/include/config.h.in \
+	$(SRCDIR)/stamp-mak.in \
+	$(SRCDIR)/stamp-h.in \
+	$(SRCDIR)/aclocal.m4 \
+	$(SRCDIR)/configure
+
+# These targets have to be updated first to not start building
+# while configure is running when any of these or their
+# dependencies change
+# Files created by configure, removing reset them the whole
+# configuration  of the source tree.
+DIST_FILES = $(OBJDIR)/config.status \
+	$(OBJDIR)/config.mak \
+	$(OBJDIR)/config.log \
+	$(OBJDIR)/stamp-mak \
+	$(OBJDIR)/stamp-h
+
+# All other files are directly specified here (see $targets).
+# removing them resets the whole build but not the configuration.
+
+# For more read "(standards) Makefile Conventions" in Info
+# or follow:
+# https://www.gnu.org/prep/standards/html_node/Makefile-Conventions.html
+
+all: $(DIST_FILES) $(OBJDIR)/include/config.h .WAIT do-all
```

Since `.WAIT` [requires make 4.4](https://lwn.net/Articles/913253/), would this work instead?
```suggestion
all: $(DIST_FILES) $(OBJDIR)/include/config.h do-all
.NOTPARALLEL: all
```

---

Or make could be told that there's a dependency, something like this?

```suggestion
all: do-first do-all
do-first: $(DIST_FILES) $(OBJDIR)/include/config.h
do-all: do-first $(targets)
```